android-sdk/src/main/java/ai/evolv/android_sdk/EvolvClientImpl.java

```diff
@@ -194,50 +178,228 @@ public void invoke(Object object) {
+                }
+
+                if (allocations.equals(JsonNull.INSTANCE)
+                        || evolvStore.config.size() == 0
+                        || allocations.getAsJsonObject().size() == 0) {
+                    return;
+                }
+
+                JsonArray entryPointEids = evolvStore.activeEntryPoints();
+
+                if (entryPointEids.size() == 0) {
+                    return;
+                }
+
+                JsonObject confirmations = new JsonObject();
+                if (remoteContext.has("confirmations")) {
+                    confirmations = remoteContext.get("confirmations").getAsJsonObject();
+                }
+
+                JsonObject confirmedCids = new JsonObject();
+                for (Map.Entry<String, JsonElement> entry : confirmations.entrySet() ) {
+                    confirmedCids = entry.getValue().getAsJsonObject().get("cid").getAsJsonObject();
+                }
+
+                JsonObject contaminations = new JsonObject();
+                if (remoteContext.has("contaminations")) {
+                    confirmations = remoteContext.get("contaminations").getAsJsonObject();
+                }
+
+                JsonObject contaminatedCids = new JsonObject();
+                for (Map.Entry<String, JsonElement> entry : contaminations.entrySet() ) {
+                    confirmedCids = entry.getValue().getAsJsonObject().get("cid").getAsJsonObject();
+                }
+
+                JsonArray confirmableAllocations = new JsonArray();
+                for (Map.Entry<String, JsonElement> alloc : allocations.getAsJsonObject().get("allocations").getAsJsonObject().entrySet()) {
+
+                    String cid = alloc.getValue().getAsJsonObject().get("cid").getAsString();
+                    String eid = alloc.getValue().getAsJsonObject().get("eid").getAsString();
+
+                    if (!confirmedCids.has(cid)
+                            && !contaminatedCids.has(cid)
+                            && hasEntryPointEids(evolvStore.activeEids, eid)) {
+                        confirmableAllocations.add(alloc.getValue());
+                    }
+                }
+
+                if (confirmableAllocations.size() == 0) {
+                    return;
+                }
+
+                long timestamp = (new Date()).getTime();
+                JsonObject contextConfirmations = new JsonObject();
+
+                for (JsonElement alloc : confirmableAllocations) {
+                    contextConfirmations.addProperty("cid", alloc.getAsJsonObject().get("cid").getAsString());
+                    contextConfirmations.addProperty("timestamp", timestamp);
+                }
+
+                contextConfirmations.add("confirmations", confirmations);
+                JsonObject newConfirmations = contextConfirmations.deepCopy();
+
+                JsonObject updateObject = new JsonObject();
+                updateObject.add("confirmations", newConfirmations);
```

Comment:
Only need to store this on 'experiments.confirmations'

Reply:
yeah, i got it